scripts/doc_checker.py

```diff
@@ -33,6 +33,19 @@
 from numpydoc.validate import Docstring
 from numpydoc.docscrape import NumpyDocString
 
+import types
+
+# fake cuDF-related modules if they're missing
+for mod_name in ("cudf", "cupy"):
```

Comment:
> can we handle the special case below by adding `"cudf.DataFrame"` to this for-loop?

well, no, you cannot `import cudf.DataFrame` because `DataFrame` is not a module... however I think that if we stub `cudf` module with a `Mock()` this probably won't be needed.

Reply:
What I think we could try to do instead is to make `doc_checker` an importable module (which it already should be now) and call its `monkeypatching()` function here in `conf.py`

Reply:
Do we want to try `mock` approach and move this to a function in this PR, or do it another, in order to do not touch and overcomplicate this big PR no more?

Reply:
I suggest creating a separate issue which we'll fix after #2944 and #3048 are closed.